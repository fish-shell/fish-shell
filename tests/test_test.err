test returned eval errors:
	invalid integer 'banana'
	in function 'is_two'
	called on standard input
	with parameter list 'banana'


test returned eval errors:
	invalid integer 'banana banana'
	in function 'is_two'
	called on standard input
	with parameter list 'banana banana'


test returned eval errors:
	invalid integer ''
	in function 'is_two'
	called on standard input


test returned eval errors:
	invalid integer 'no one'
	in function 'is_two'
	called on standard input
	with parameter list 'no one'


test returned eval errors:
	invalid integer 'everyone'
	in function 'is_two'
	called on standard input
	with parameter list 'everyone'


test returned eval errors:
	invalid integer '0b10'
	in function 'is_two'
	called on standard input
	with parameter list '0b10'


