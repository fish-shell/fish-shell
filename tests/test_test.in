# Tests for the `test` builtin. TEST!
function is_two
    # TEST?
    if test 2 -eq "$argv"
        echo argv is two
    else
        echo argv is not two
    end
end

is_two 5 # This is indeed not two
is_two 2 # This fine specimen, on the other hand, is.
is_two " 2 " # This is also two.

# Consider the banana. It is not two.
is_two banana
is_two banana banana # While they are two bananas, they are not two itself.
is_two # When no one is there to be two, is it still two?
is_two "no one" # No one is not two.
is_two everyone # is also not two.

is_two 0b10
is_two (printf '%d\n' 0x2) # Twobular!
