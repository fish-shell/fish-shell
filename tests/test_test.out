argv is not two
argv is two
argv is two
argv is not two
argv is not two
argv is not two
argv is not two
argv is not two
argv is not two
